### PR TITLE
Feedback screen Implementation

### PR DESCRIPTION
**Pull request by giurassic**

### CODE REVIEW

File: app/src/main/java/berlin/eloquent/eloquentandroid/di/main/MainModule.kt [hunk 1]
```diff
@@ -13,8 +14,8 @@ object MainModule {
     @MainScope
     @Provides
     @JvmStatic
-    fun provideRecorderRepository(dao: RecordingDao): IRecorderRepository {
-        return RecorderRepository(dao)
+    fun provideRecorderRepository(dao: RecordingDao,audioService:AudioService): IRecorderRepository {
```
Comment (heymage) [inline app/src/main/java/berlin/eloquent/eloquentandroid/di/main/MainModule.kt]:
```suggestion
    fun provideRecorderRepository(dao: RecordingDao, audioService: AudioService): IRecorderRepository {
```

File: app/src/main/java/berlin/eloquent/eloquentandroid/di/main/MainModule.kt [hunk 2]
```diff
@@ -13,8 +14,8 @@ object MainModule {
...
+        return RecorderRepository(dao,audioService)
```
Comment (heymage) [inline app/src/main/java/berlin/eloquent/eloquentandroid/di/main/MainModule.kt]:
```suggestion
        return RecorderRepository(dao, audioService)
```

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt [hunk 3]
```diff
@@ -39,7 +45,23 @@ class FeedbackFragment : Fragment() {
         val args: FeedbackFragmentArgs by navArgs()
         viewModel.setRecording(args.recordingId)
 
+//        binding.btnGetAnalysis.setOnClickListener{
+//            print("Called 2")
+//            viewModel.getData()
```
Comment (heymage) [inline app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt]:
```suggestion
```

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt [hunk 4]
```diff
@@ -39,7 +45,23 @@ class FeedbackFragment : Fragment() {
         val args: FeedbackFragmentArgs by navArgs()
         viewModel.setRecording(args.recordingId)
 
+//        binding.btnGetAnalysis.setOnClickListener{
```
Comment (heymage) [inline app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt]:
```suggestion
```

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt [hunk 5]
```diff
@@ -39,7 +45,23 @@ class FeedbackFragment : Fragment() {
...
+//            print("Called 2")
```
Comment (heymage) [inline app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt]:
```suggestion
```

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt [hunk 6]
```diff
@@ -39,7 +45,23 @@ class FeedbackFragment : Fragment() {
...
+//            viewModel.getData()
+//        }
```
Comment (heymage) [inline app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt]:
```suggestion
```

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/retrofit/DataModel.kt [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package berlin.eloquent.eloquentandroid.main.feedback.retrofit
+
+import androidx.annotation.Keep
+
+@Keep
+class DataModel {
+    var wordsPerSecond: String? = null
+}
```
Comment (heymage) [inline app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/retrofit/DataModel.kt]:
```suggestion
}

```

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/repository/RecorderRepository.kt [hunk 8]
```diff
@@ -49,4 +53,10 @@ class RecorderRepository @Inject constructor(
         }
     }
 
+    override suspend fun getAnalysis(): Response<DataModel> {
+        return withContext(ioDispatcher) {
+            return@withContext service.getAnalysis()
+        }
+    }
+
 }
```
Comment (heymage) [inline app/src/main/java/berlin/eloquent/eloquentandroid/main/repository/RecorderRepository.kt]:
```suggestion
}

```

File: .idea/jarRepositories.xml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (heymage) [inline .idea/jarRepositories.xml:1]: Untrack this file in your git, it's a generated file and gets regenerated all the time.

**Review by heymage — CHANGES_REQUESTED** [review]
There are some issues, please have a look on it. Thanks 🥇

File: app/src/main/java/berlin/eloquent/eloquentandroid/di/main/MainModule.kt (hunk 1 above)
Comment (heymage) [reply app/src/main/java/berlin/eloquent/eloquentandroid/di/main/MainModule.kt]: This is for keeping a consistent code style

File: app/src/main/java/berlin/eloquent/eloquentandroid/di/main/MainModule.kt (hunk 2 above)
Comment (heymage) [reply app/src/main/java/berlin/eloquent/eloquentandroid/di/main/MainModule.kt]: This is for keeping a consistent code style

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt (hunk 3 above)
Comment (heymage) [reply app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt]: remove comments in code which should get merged

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt (hunk 4 above)
Comment (heymage) [reply app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt]: remove comments in code which should get merged

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt (hunk 5 above)
Comment (heymage) [reply app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt]: remove comments in code which should get merged

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt (hunk 6 above)
Comment (heymage) [reply app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/FeedbackFragment.kt]: remove comments in code which should get merged

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/retrofit/DataModel.kt (hunk 7 above)
Comment (heymage) [reply app/src/main/java/berlin/eloquent/eloquentandroid/main/feedback/retrofit/DataModel.kt]: new lines should be at the end of each file

File: app/src/main/java/berlin/eloquent/eloquentandroid/main/repository/RecorderRepository.kt (hunk 8 above)
Comment (heymage) [reply app/src/main/java/berlin/eloquent/eloquentandroid/main/repository/RecorderRepository.kt]: new lines should be at the end of each file

Comment (giurassic) [reply app/src/main/java/berlin/eloquent/eloquentandroid/main/repository/RecorderRepository.kt]: For a specific reason or just for the looks ?

File: .idea/jarRepositories.xml (hunk 9 above)
Comment (giurassic) [reply .idea/jarRepositories.xml:1]: will do

**Pull request merged**